contracts/base/ModuleManager.sol

```diff
@@ -152,11 +152,8 @@ abstract contract ModuleManager is SelfAuthorized, Executor, IModuleManager {
         uint256 value,
         bytes memory data,
         Enum.Operation operation
-    ) public virtual override returns (bool success) {
-        (address guard, bytes32 guardHash) = preModuleExecution(to, value, data, operation);
-
-        success = execute(to, value, data, operation, type(uint256).max);
-        postModuleExecution(guard, guardHash, success);
+    ) public override returns (bool success) {
+        return _onExecTransactionFromModule(to, value, data, operation);
```

Comment:
nit: What I meant in the sync was not to move the whole function to an `_onExecTransactionFromModule` (this increases code size because of the duplicated event emitting code), but instead refactor the contracts slightly to include an `_onAfterExecTransactionFromModule` that emits the extra event in L2.

I also think that for consistency, we should do the same for `execTransaction` function.

Reply:
contract already has `postModuleExecution` and adding another function `_onAfterExecTransactionFromModule` with similar semantics would create confusion I think. And if we have a single post exec function then, `SafeL2` has to be override it to emit event.

Reply:
> In fact, I think _onAfterExecTransactionFromModule should be called in postModuleExecute.

I was also thinking same. That sounds more reasonable.

Reply:
added here: https://github.com/safe-global/safe-smart-account/pull/772/commits/bee704b8f940b59ae4b627b4a28107c5a393a032